install/kubernetes/helm/istio/templates/configmap.yaml

```diff
@@ -83,13 +83,26 @@ data:
       # If set to 0 (default), then start worker thread for each CPU thread/core.
       concurrency: {{ .Values.global.proxy.concurrency }}
       #
-      # Zipkin trace collector
-      {{- if .Values.global.tracer.zipkinAddress }}
-      zipkinAddress: {{ .Values.global.tracer.zipkinAddress }}
+      tracing:
+      {{- if eq .Values.global.proxy.tracer "lightstep" }}
+        lightstep:
+          # Address of the LightStep Satellite pool
+          address: {{ .Values.global.tracer.lightstep.address }}
+          # Access Token used to communicate with the Satellite pool
+          accessToken: {{ .Values.global.tracer.lightstep.accessToken }}
+          # Whether communication with the Satellite pool should be secure
+          secure: {{ .Values.global.tracer.lightstep.secure }}
+          # Path to the file containing the cacert to use when verifying TLS
+          cacertPath: {{ .Values.global.tracer.lightstep.cacertPath }}
+      {{- else if eq .Values.global.proxy.tracer "zipkin" }}
```

Comment:
To remain backwards compatible with the current behaviour, would prefer that the `.Values.global.proxy.tracer` must have a value, and by default it should be `zipkin`. Then within this branch, have another if/else clause based on whether the `.Values.global.tracer.zipkin.address` is defined.

The same approach should be used in the remote configmap.

Reply:
@objectiser I really like the suggestion to make `.Values.global.proxy.tracer` default to `zipkin` and I'll definitely add that in. Likewise I'll move setting the default zipkin address into this conditional block.

I would like to push back on the notion that `.Values.global.proxy.tracer` "must" have a value though. The existing behavior makes it such that there is always a zipkinAddress set which (according to my understanding) means that the bootstrap for the proxies will always be set up with a zipkin cluster. If we allow `.Values.global.proxy.tracer` to be unset, then we gain the ability to omit that from the bootstrap config while not changing the default behavior.

In the end I'm happy to defer to your judgement on this.

Reply:
WE set the zipkin address so that we can generate the tracing configuration in Envoy. (which cannot be changed through anything other than bootstrap). But we don't enable tracing unless the user sets the enableTracing to true. I think we can set the appropriate things in envoy through lds/rds IIRC. 

I don't care either way.. leave it to you guys.

Reply:
Don't have a strong opinion either way. We just need to make sure that unsetting the `.Values.global.proxy.tracer` doesn't have any adverse effects.